src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -53,70 +47,47 @@ public ExpenseOD addExpenseInMonth(ExpenseID expenseIDAdd) throws EntityExceptio
         ArrayList<Category> categoryList = month.getCategoryData();
         this.recurringExpenseList = session.getRecurData();
 
-        try {
-            double valueDouble = toDouble(expenseIDAdd.getValue());
-
-            Category selectedCategory = findCategory(categoryList, expenseID.getOldCategory());
-
+        try {double valueDouble = toDouble(expenseIDAdd.getValue());
+            Category selectedCategory = findCategory(categoryList, expenseID.getCategory());
             if (valueDouble < 0) {
-                // 2. Expense value less than 0 fail
+                //Expense value less than 0 fail: When a user tries to add the expense value with a negative number.
                 ExpenseOD expenseODFailAdd = new ExpenseOD("Expense value can't be less than $0. Please try again!");
-                return expenseOB.fail(expenseODFailAdd);
-            }
+                return expenseOB.fail(expenseODFailAdd);}
             if (expenseID.getIsRecurringExpense()) {
                 if(recurringExpenseList.size() > 0) {
-                    for (Expense expense1 : recurringExpenseList) {
+                    for (Expense expense1 : monthExpenseList) {
```

Comment:
This for loop should be above line 56. Currently it is only checking for if an expense has the same name if the expense is recurring and there are recurring expenses, which should not happen. Additionally, you should get rid of the if statement in line 57, because it prevents recurring expenses from being added if no recurring expenses exist, which means the user would never be able to add a recurring expense.